runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpServerFactory.java

```diff
@@ -4847,11 +4877,23 @@ else if (!isCorsRequestAllowed(binding, headers))
                                     .headers(hs -> headers.forEach((n, v) -> hs.item(h -> h.name(n).value(v))))
                                     .build();
 
-                            exchange.doRequestBegin(traceId, beginEx);
+                            HttpRequestType request = binding.resolveRequest(headers::get);
```

Comment:
Try not to lean too heavily on the map of headers we create here as it is undesirable and we'll likely remove it as some point..

Reply:
fixed to not use the map, pls chk